### ISSUE TITLE
Add conn.interrupt() to DuckDB python API

### ISSUE THREAD

**Issue by Mytherin** [duckdb/duckdb#5938]
### Discussed in https://github.com/duckdb/duckdb/discussions/5937

SQLite supports `conn.interrupt` - we have support for this in the C++ client, we should add support for this in the Python client as well.

**Comment by swhite2** [comment]
@Mytherin Hi there, it just so happens I was looking for this feature today. Though maybe you could enlighten me as to whether I even need this.

Suppose I have the following situation:

- a python select() call blocks until a file object becomes ready. If I'd want to terminate this immediately on a SIGTERM, a signal handler would be necessary in which we raise an exception, the exception is necessary to prevent a default retry of the select() syscall (PEP475), so we can return immediately.
- now suppose the file object becomes ready, and we start either an `append()` (DataFrame) `INSERT` or `UPDATE`. During one of these operations, a SIGTERM comes in, and the exception handler raises an `InterruptedError` as intended, handing the execution flow back to my signal handler and leaving the duckdb action in an unknown state.
- Also assume that all of this happens within a custom context manager that always properly calls `close()` on the duckdb connection on any exception or by going out of scope.

In short: the above DB operations could be interrupted and immediately followed by a `close()`.

Can I assume the transaction is automatically rolled back?

Many thanks in advance!

**Comment by Mytherin** [comment]
Yes, if a transaction is aborted due to e.g. an interrupt the transaction will be rolled back in its entirety.

**Comment by swhite2** [comment]
Thanks!

**Comment by henrinikku** [comment]
Hey, it's my first time contributing and this seems like a good place to start. I'll submit a PR when ready.
